gyp/pylib/gyp/generator/msvs.py

```diff
@@ -294,20 +302,29 @@ def _ConfigFullName(config_name, config_data):
   return '%s|%s' % (_ConfigBaseName(config_name, platform_name), platform_name)
 
 
-def _ConfigWindowsTargetPlatformVersion(config_data):
-  ver = config_data.get('msvs_windows_target_platform_version')
-  if not ver or re.match(r'^\d+', ver):
-    return ver
-  for key in [r'HKLM\Software\Microsoft\Microsoft SDKs\Windows\%s',
-              r'HKLM\Software\Wow6432Node\Microsoft\Microsoft SDKs\Windows\%s']:
-    sdkdir = MSVSVersion._RegistryGetValue(key % ver, 'InstallationFolder')
-    if not sdkdir:
-      continue
-    version = MSVSVersion._RegistryGetValue(key % ver, 'ProductVersion') or ''
-    # find a matching entry in sdkdir\include
-    names = sorted([x for x in os.listdir(r'%s\include' % sdkdir) \
-                    if x.startswith(version)], reverse = True)
-    return names[0]
+def _ConfigWindowsTargetPlatformVersion(config_data, version):
```

Comment:
```suggestion
def _ConfigWindowsTargetPlatformVersion(config_data, version):
  target_ver = config_data.get('msvs_windows_target_platform_version')
  if target_ver and re.match(r'^\d+', target_ver):
    return target_ver
```

@targos sorry for the delay getting to this. This PR always fails when using VS2017 (not only Windows 2016). I see `msvs_windows_target_platform_version` was renamed to `msvs_windows_sdk_version`. However, the semantics here changed a bit so simply renaming in `lib/configure.js` does not work. I suggest to bring back the old variable name with old semantics, if this passes in CI. I'm not sure how well this works outside node-gyp/node, but I don't know if that's relevant.